types/src/bytesrepr.rs

```diff
@@ -1317,6 +1087,58 @@ where
     }
 }
 
+/// Serializes dynamic bytes array.
```

Comment:
I think we should be a_lot_ more explicit in the comments here, since this function has the same signature as `serialize_array`.  However, I'd rather see the signature change anyway (see following comment).  The docs should also say _why_ the function is needed, and that we'll remove it once trait specialization lands.

Same goes for the following six functions.